akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java

```diff
@@ -341,9 +342,13 @@ public RetentionCriteria retentionCriteria() {
       // this makes the context available to the command handler etc.
       private final ActorContext<Command> ctx;
 
+      // optionally if you need `ActorContext.getSelf()`
```

Comment:
Perhaps
```suggestion
      // optionally if you only need `ActorContext.getSelf()`
```
since in this example we're already keeping `ctx` around.

Is keeping `ctx` around like this valid, actually?

Reply:
yes, it's completely valid and it's the approach we have taken for all `EventSourcedBehavior` and `AbstractBehavior`. Lately also in javadsl functional style: https://github.com/akka/akka/pull/27121